src/sentry/search/events/builder/metrics.py

```diff
@@ -151,7 +151,7 @@ def load_config(self) -> DatasetConfig:
             return super().load_config()
 
         if self.dataset in [Dataset.Metrics, Dataset.PerformanceMetrics]:
-            if self.builder_config.use_metrics_layer:
+            if self.use_metrics_layer:
```

Comment:
Seems like `load_config` was using the `use_metrics_layer` attribute directly from the builder config. Not sure if there is a specific reason for this, but I had to update it to use the `self.use_metrics_layer` property which has additional checks